lib/active_hash/base.rb

```diff
@@ -205,7 +163,9 @@ def create!(attributes = {})
       end
 
       def all(options = {})
-        ActiveHash::Relation.new(self, @records || [], options[:conditions] || {})
+        relation = ActiveHash::Relation.new(self, @records || [])
+        relation = relation.where!(options[:conditions]) if options[:conditions]
+        relation
```

Comment:
is there a reason to change this block?

Reply:
I suppose the `Relation` could be initialized with the conditions directly.